src/main/java/io/reactivex/subscribers/TestSubscriber.java

```diff
@@ -795,16 +795,27 @@ public final boolean awaitTerminalEvent(long duration, TimeUnit unit) {
      * @return this
      */
     public final TestSubscriber<T> assertFusionMode(int mode) {
-        if (establishedFusionMode != mode) {
+        int m = establishedFusionMode;
+        if (m != mode) {
             if (qs != null) {
-                throw new AssertionError("Fusion mode different. Expected: " + mode + ", actual: " + establishedFusionMode);
+                throw new AssertionError("Fusion mode different. Expected: " + fusionModeToString(mode)
+                + ", actual: " + fusionModeToString(m));
             } else {
                 throw new AssertionError("Upstream is not fuseable");
             }
         }
         return this;
     }
     
+    private String fusionModeToString(int mode) {
+        switch (mode) {
+        case QueueSubscription.NONE : return "NONE";
+        case QueueSubscription.SYNC : return "SYNC";
+        case QueueSubscription.ASYNC : return "ASYNC";
+        default: return "Unknown(" + mode + ")";
```

Comment:
well, exception will be thrown not here, but in `assertFusionMode()` and only if fusion mode will be not expected. Imagine if you'll add another fusion mode, `fusionModeToString()` will return `Unknown(m)` which is kind of weird